test/python/transpiler/test_pass_scheduler.py

```diff
@@ -550,6 +550,158 @@ def test_fresh_initial_state(self):
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassA_TP_NR_NP",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 6",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 6 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassA_TP_NR_NP",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 5",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 5 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassA_TP_NR_NP",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 4",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 4 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 3",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 3 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 2",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 2 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 2",
+            "run analysis pass PassG_calculates_dag_property",
+            "set property as 2 (from dag.property)",
+            "run analysis pass PassK_check_fixed_point_property",
+            "run transformation pass PassF_reduce_dag_property",
+            "dag property = 2",
+        ]
+        self.assertScheduler(self.circuit, self.passmanager, expected)
+
+    def test_multiple_nested_conditionals(self):
+        """Run multiple nested conditionals."""
+        nested_conditional1 = [
+            ConditionalController(
+                [PassB_TP_RA_PA()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+        nested_conditional2 = [
+            ConditionalController(
+                [PassC_TP_RA_PA()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+        nested_conditional3 = [
+            ConditionalController(
+                [PassD_TP_NR_NP()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+
+        self.passmanager.append(
+            [PassE_AP_NR_NP(True)]
+            + nested_conditional1
+            + [PassE_AP_NR_NP(False)]
+            + nested_conditional2
```

Comment:
I don't think this tests the feature that is added in this PR. It is in line with other tests in this file. But, it doesn't belong with this PR. In particular, the conditional passes run sequentially, rather than nested-ly.

Reply:
I removed this test in a commit.